lib/iris/tests/unit/concatenate/test_concatenate.py

```diff
@@ -261,147 +230,156 @@ def test_asc_points(self):
-        self.assertEqual(result1, result2)
+        assert len(result1) == 1
+        assert result1 == result2
 
     def test_desc_bounds_all_singleton(self):
         top = self._make_cube([5], [10, 0])
         bottom = self._make_cube([15], [20, 10])
         result1 = concatenate([top, bottom])
         result2 = concatenate([bottom, top])
-        self.assertEqual(len(result1), 1)
-        self.assertEqual(len(result2), 1)
-        self.assertEqual(result1, result2)
+        assert len(result1) == 1
+        assert result1 == result2
 
 
-class TestConcatenate__dask(tests.IrisTest):
-    def build_lazy_cube(self, points, bounds=None, nx=4, aux_coords=False):
+class TestConcatenate__dask:
+    @staticmethod
+    def build_lazy_cube(points):
+        nx = 4
         data = np.arange(len(points) * nx).reshape(len(points), nx)
         data = as_lazy_data(data)
         cube = iris.cube.Cube(data, standard_name="air_temperature", units="K")
-        lat = iris.coords.DimCoord(points, "latitude", bounds=bounds)
+        lat = iris.coords.DimCoord(points, "latitude")
         lon = iris.coords.DimCoord(np.arange(nx), "longitude")
         cube.add_dim_coord(lat, 0)
         cube.add_dim_coord(lon, 1)
-        if aux_coords:
-            bounds = np.arange(len(points) * nx * 4).reshape(len(points), nx, 4)
-            bounds = as_lazy_data(bounds)
-            aux_coord = iris.coords.AuxCoord(data, var_name="aux_coord", bounds=bounds)
-            cube.add_aux_coord(aux_coord, (0, 1))
         return cube
 
-    def test_lazy_concatenate(self):
+    @pytest.fixture()
+    def sample_lazy_cubes(self):
+        # Make a pair of concatenatable cubes, with dim points [1, 2] and [3, 4, 5]
         c1 = self.build_lazy_cube([1, 2])
         c2 = self.build_lazy_cube([3, 4, 5])
-        (cube,) = concatenate([c1, c2])
-        self.assertTrue(cube.has_lazy_data())
-        self.assertFalse(ma.isMaskedArray(cube.data))
+        return c1, c2
+
+    @staticmethod
+    def add_sample_auxcoord(cube):
+        # Augment a test cube by adding an aux-coord on the concatenation dimension
+        n_points, nx = cube.shape
+        bounds = np.arange(n_points * nx * 4).reshape(n_points, nx, 4)
+        bounds = as_lazy_data(bounds)
+        aux_coord = iris.coords.AuxCoord(
+            cube.core_data(),
+            bounds=bounds,
+            var_name="aux_coord",
+        )
+        cube.add_aux_coord(aux_coord, (0, 1))
 
-    def test_lazy_concatenate_aux_coords(self):
-        c1 = self.build_lazy_cube([1, 2], aux_coords=True)
-        c2 = self.build_lazy_cube([3, 4, 5], aux_coords=True)
+    def test_lazy_concatenate(self, sample_lazy_cubes):
+        (cube,) = concatenate(sample_lazy_cubes)
+        assert cube.has_lazy_data()
+        assert not ma.isMaskedArray(cube.data)
+
+    def test_lazy_concatenate_aux_coords(self, sample_lazy_cubes):
+        c1, c2 = sample_lazy_cubes
+        c2 = self.build_lazy_cube([3, 4, 5])
```

Comment:
@pp-mo Why the need to call `build_lazy_cube` here?

Isn't this the same as `sample_lazy_cubes[1]` ?

Reply:
NO, I think I was missing something obvious!
I guess just because I didn't initially equate these with the "standard setup", due to use of "aux_coords=True" in the original.

Reply:
And in that case, "build_lazy_cubes" doesn't need to be provided separately at all ...

Reply:
https://github.com/SciTools/iris/pull/5813/commits/2427de8a3b93c22fa95ee4b5c1a67b42e2f8a560